File: 657-robot-return-to-origin/robot-return-to-origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        int upper = 0;
        int lower = 0;
        int right = 0;
        int left = 0;

        for (char ch : moves.toCharArray()){
            if (ch == 'U')upper++;
            else if(ch == 'D')lower++;
            else if(ch == 'L')left++;
            else{
                right++;
            }
        }

        if(Math.abs(upper-lower) != 0 || Math.abs(right-left) != 0){
            return false;
        }

        return true;
        
    }
}

File: 657-robot-return-to-origin/robot-return-to-origin.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        hsh={
            "R" : 0,
            "L" : 0,
            "U" : 0,
            "D" : 0
        }

        for i in moves:
            hsh[i]+=1
        
        if abs(hsh["R"] - hsh["L"]) != 0:
            return False
        else:
            if abs(hsh["U"]-hsh["D"]) != 0:
                return False
        return True
        